### PR TITLE
move apparmor annotation constants to k8s.io/api/core/v1

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

**What type of PR is this?**
/kind api-change

**What this PR does / why we need it**:
Noticed the TODO to move apparmor constants to an api package while working on https://github.com/kubernetes/kubernetes/issues/74352.  

**Which issue(s) this PR fixes**:
Needed for https://github.com/kubernetes/kubernetes/issues/74352

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/api/core/v1/annotation_key_constants.go [hunk 1]
```diff
@@ -48,6 +48,22 @@ const (
...
 
+	// AppArmorBetaContainerAnnotationKeyPrefix is the prefix to an annotation key specifying a container's apparmor profile.
+	AppArmorBetaContainerAnnotationKeyPrefix = "container.apparmor.security.beta.kubernetes.io/"
+	// AppArmorBetaDefaultProfileAnnotatoinKey is the annotation key specifying the default AppArmor profile.
+	AppArmorBetaDefaultProfileAnnotationKey = "apparmor.security.beta.kubernetes.io/defaultProfileName"
+	// AppArmorBetaAllowedProfileAnnotationKey is the annotation key specifying the allowed AppArmor profiles.
+	AppArmorBetaAllowedProfilesAnnotationKey = "apparmor.security.beta.kubernetes.io/allowedProfileNames"
+
+	// AppArmorProfileRuntimeDefault is the profile specifying the runtime default.
+	AppArmorProfileRuntimeDefault = "runtime/default"
```
Comment (andrewsykim) [inline staging/src/k8s.io/api/core/v1/annotation_key_constants.go]: Maybe `AppArmorProfileRuntimeDefault`, `AppArmorProfileNamePrefix` and  `AppArmorProfileNameUnconfined` belong in a separate file, but kept it here to be consistent with `SeccompProfileRuntimeDefault`

File: pkg/api/pod/util.go [hunk 2]
```diff
@@ -19,11 +19,11 @@ package pod
 import (
 	"strings"
 
+	"k8s.io/api/core/v1"
```
Comment (neolit123) [inline pkg/api/pod/util.go]: i don't think we can do that. the apparmor feature is considered Beta still:
https://kubernetes.io/docs/tutorials/clusters/apparmor/

can be possibly moved under a separate package, somewhere under:
`k8s.io/api`

cc @tallclair 
/sig auth

Comment (andrewsykim) [reply pkg/api/pod/util.go]: My reasoning was that the seccomp constants are also beta but they live in k8s.io/api/core/v1. Happy to move them somewhere else, but this seemed like the most appropriate home at the moment.

Comment (neolit123) [reply pkg/api/pod/util.go]: ack, something was telling me that we already have <GA constants in v1.

Comment (liggitt) [reply pkg/api/pod/util.go]: The annotations target core/v1 objects (pods), and are already validated in core/v1 validation, so this is probably the right package for them.

File: staging/src/k8s.io/api/core/v1/annotation_key_constants.go (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/api/core/v1/annotation_key_constants.go]: It's unclear these values will be used verbatim in the final promoted fields. Suggest naming the constants (and adding godoc) to indicate they pair with the beta annotations, e.g.:

```go
AppArmorBetaAnnotationProfileRuntimeDefault
AppArmorBetaAnnotationProfileNamePrefix
AppArmorBetaAnnotationProfileNameUnconfined
```

Comment (andrewsykim) [reply staging/src/k8s.io/api/core/v1/annotation_key_constants.go]: done

**Comment by liggitt** [comment]
/approve
/lgtm

needs rebase

**Pull request merged**